source/common/stats/symbol_table_impl.cc

```diff
@@ -205,23 +253,47 @@ bool SymbolTable::lessThan(const StatName& a, const StatName& b) const {
   // If this becomes a performance bottleneck (e.g. during sorting), we could
   // provide an iterator-like interface for incrementally decoding the symbols
   // without allocating memory.
-  SymbolVec av = SymbolEncoding::decodeSymbols(a.data(), a.numBytes());
-  SymbolVec bv = SymbolEncoding::decodeSymbols(b.data(), b.numBytes());
+  SymbolVec av = SymbolEncoding::decodeSymbols(a.data(), a.dataSize());
+  SymbolVec bv = SymbolEncoding::decodeSymbols(b.data(), b.dataSize());
   for (uint64_t i = 0, n = std::min(av.size(), bv.size()); i < n; ++i) {
     if (av[i] != bv[i]) {
       Thread::LockGuard lock(lock_);
-      return fromSymbol(av[i]) < fromSymbol(bv[i]);
+      bool ret = fromSymbol(av[i]) < fromSymbol(bv[i]);
+      return ret;
     }
   }
   return av.size() < bv.size();
 }
 
+#ifndef ENVOY_CONFIG_COVERAGE
+void SymbolTable::debugPrint() const {
+  Thread::LockGuard lock(lock_);
+  std::vector<Symbol> symbols;
+  for (const auto& p : decode_map_) {
+    symbols.push_back(p.first);
+  }
+  std::sort(symbols.begin(), symbols.end());
+  for (Symbol symbol : symbols) {
+    std::string& token = *decode_map_.find(symbol)->second;
+    const SharedSymbol& shared_symbol = encode_map_.find(token)->second;
+    ENVOY_LOG_MISC(info, "{}: '{}' ({})", symbol, token, shared_symbol.ref_count_);
+  }
+}
+#endif
+
 StatNameStorage::StatNameStorage(absl::string_view name, SymbolTable& table) {
   SymbolEncoding encoding = table.encode(name);
   bytes_ = std::make_unique<uint8_t[]>(encoding.bytesRequired());
   encoding.moveToStorage(bytes_.get());
```

Comment:
Since moveToStorage() returns a number of bytes transferred, would it make sense to non-release assert something about the return value here?

Reply:
I don't think it would be that interesting an assert; moveToStorage obviously returns a positive number as it includes the length.